src/app/public/test-utility/test-utility.spec.ts

```diff
@@ -38,6 +45,63 @@ class TestParentComponent { }
 })
 class TestComponent { }
 
+@Component({
+  selector: 'test-overlay-internal-cmp',
+  template: `
+    <div id="my-modal-content">
+      foobar
+    </div>
+    <ul>
+      <li class="my-item">foo</li>
+      <li class="my-item">bar</li>
+      <li class="my-item">baz</li>
+    </ul>
+  `
+})
+class TestOverlayInternalComponent { }
+
+@Component({
+  selector: 'test-overlay-cmp',
+  template: `
+  <button
+    class="sky-btn sky-btn-primary sky-margin-inline-default"
+    type="button"
+    (click)="launchOverlay()"
+  >
+    Launch overlay
+  </button>
+`
+})
+class TestOverlayComponent {
+
+  public overlays: SkyOverlayInstance;
+
+  constructor(
+    public overlayService: SkyOverlayService
+  ) { }
+
+  public launchOverlay(): void {
+    const overlayInstance = this.overlayService.create({});
+
+    const componentInstance = overlayInstance.attachComponent(
+      TestOverlayInternalComponent
+    );
+
+    this.overlays = overlayInstance;
+  }
+}
+
+@NgModule({
+  declarations: [
+    TestOverlayComponent,
+    TestOverlayInternalComponent
+  ],
+  entryComponents: [
+    TestOverlayInternalComponent
+  ]
+})
+class OverlayTestModule {}
```

Comment:
☝️  I considered breaking these out into their own file... There were already a couple test components in here so I left them in. I could be persuaded if the reviewer thinks its worth it.